### ISSUE TITLE
Parent `hx-vals` overriding child value/input

### ISSUE THREAD

**Issue by jreviews** [bigskysoftware/htmx#433]
In the following example, `large` and `blue` are always sent. 

```html
<form hx-get="/test" hx-vals='{"size": "large", "color": "blue"}'>    
    <button hx-vals='{"size": "small"}'>Button</button>
    <input type="text" name="color" value="red">
</form>
```

**Comment by 1cg** [comment]
@bencroker any opinions on this one?

**Comment by jreviews** [comment]
I found it a bit counter-intuitive and one instance where htmx's parent element inheritance of child attributes seems to work a bit different.

**Comment by bencroker** [comment]
I tend to disagree. If you were to do this and click the button, it would work as you are expecting because the button is the child.

```
<form hx-get="/test" hx-vals='{"size": "large", "color": "blue"}'>    
    <button hx-get="/test" hx-vals='{"size": "small"}'>Button</button>
    <input type="text" name="color" value="red">
</form>
```

**Comment by bencroker** [comment]
Well, if you were to POST the form then it might work differently. The parse order of the variables is one of those things that gets really complex in htmx.

**Comment by jreviews** [comment]
Just an FYI for future reference, posting the form doesn't change the results. That was just an example, but the behavior is similar when not using a form tag.

**Comment by jarrodmoldrich** [comment]
@bencroker I've experienced this same behaviour when adding a `hx-vals` to a child with a `hx-trigger` that `consume`s.  After fumbling about I found that explicitly adding a hx-post/get fixed the problem.

So just to know I've got this right, this is because the hx-post/get is what's considered the originator of the event?  And `hx-trigger` just adds an `hx-include` to the closest ancestor with a post/get?  I just blithely assumed that a `hx-trigger` with a `consume` modifier meant that it's only that element that initiates the request, and so should override the values.

Might it be worth adding some documentation/diagrams about how overrides/requests work for dullards like myself? 😅
